### ISSUE TITLE
Reskin&Oldskin - After the repayment recovery of written-off loan, the "recovery repayment" button is still available

### ISSUE THREAD

**Issue by MexinaD** [openMF/community-app#2225]
When you write-off the loan, the loan appear with the button for repayment recovery
When you recover the repayment of the whole loan amount, the button should be removed.
But for now, the button is still there which lead for a user to be able to do many recovery more than the loan amount 

In the reskined
![recovery](https://cloud.githubusercontent.com/assets/8160209/25945356/3d9d7442-364f-11e7-9393-6fb99ecf9a5d.png)

In community-app
![recovery1](https://cloud.githubusercontent.com/assets/8160209/25945367/497f460a-364f-11e7-85ea-d76a635f19dc.png)

**Comment by Mastermaulik** [comment]
@mbj36 Please look into this issue

**Comment by Mastermaulik** [comment]
@botraunak Can you help?

**Comment by gkrishnan724** [comment]
@MexinaD @santoshmath What should be there after the repayment has been recovered? Should the button be hidden?

**Comment by santoshmath** [comment]
@gkrishnan724 , Yes 'Recovery Repayment' button should be hidden only after the required amount is recovered. Sometimes, recovery repayment is done in multiple times rather than one time. Hiding 'Recovery Repayment' should check whether all the amount is recovered or not.

**Comment by edcable** [comment]
@bharathcgowda can you confirm the behavior as noted by @santoshconflux

**Comment by bharathcgowda** [comment]
@edcable behavior is confirmed.
loan example for reference
https://staging.mifos.io/#/viewloanaccount/99

**Comment by luckyman20** [comment]
@edcable I am taking up this issue. I have [raised a ticket](https://issues.apache.org/jira/projects/FINERACT/issues/FINERACT-1113) on Fineract to send the amount of payment recovered when the loan is written off.

I will send a Pull Request just after we get a patch in Fineract.

**Comment by tonic889** [comment]
I would like to take this up on the fineract side.

**Comment by luckyman20** [comment]
Sure @tonic889

**Comment by tonic889** [comment]
@luckyman20 Why is there a fineract-side change required for this? It seems to me it should be possible to hide the button when the Current Balance is 0, correct?  This could be done in ViewLoanDetailsController.js in the community app without any fineract side changes.  Please correct me if I'm wrong.

**Comment by bharathcgowda** [comment]
Hi @tonic889 , when you do a write-off, the current balance and current outstanding will be 0. Hence we cannot use any of the existing fields.  Besides the right logic is to verify the total "written off amount" with the "total recovered amount", when the "total recovered amount" becomes equal to the "total written off amount", then it makes operational sense to hide the "Recover" button.


![image](https://user-images.githubusercontent.com/65952533/90592290-ddf73880-e202-11ea-90f3-c071abb5551b.png)

**Comment by tonic889** [comment]
Ah, yes. Ok, I will take it up soon.

**Comment by luckyman20** [comment]
@tonic889 Let us know if you are still interested to take this up, I know I promised to work on this but I have been swamped with lot of work lately.

**Comment by tonic889** [comment]
@luckyman20 Yes, I can take this up in the next few days.

**Comment by tonic889** [comment]
@luckyman20 @bharathcgowda I need some clarification here.  What I observed is that when I enter a recovery payment, the total written off is reduced by the recovery payment amount.  Therefore, the logic stated earlier that the button should be hidden when total recovered >= total waived would not be correct.  I believe the desired logic is the button should be hidden when the total written off amount = 0.  Please let me know if you agree and I will make the change.  Thanks.

Another thing I am wondering is whether the recovery payment screen itself should have some validation to ensure the user can not enter a recovery payment > the written off amount? It seems currently I can enter any amount for the recovery payment.

**Comment by bharathcgowda** [comment]
@tonic889  That sounds good to me Brian, we can go ahead with the proposed logic.

**Comment by tonic889** [comment]
Community app PR #3310 and Fineract pull request #1356 submitted for this issue.
